synthesizer/program/src/resources/credits.aleo

```diff
@@ -31,6 +31,18 @@ struct committee_state:
     microcredits as u64;
     // The boolean flag indicating if the validator is open to new stakers.
     is_open as boolean;
+    // The percent (0-100) of delegator commissions which validator retains.
+    // Set by validator on self bond.
+    commission as u8;
```

Comment:
You need a snarkVM change to actually change how rewards are calculated and distributed in consensus as it's not done as part of the credits program.

Reply:
Great call out. I 100% agree. Should have called that out in the description explicitly. As the main focus of this was ARC-41 I decided to focus there and not attempt VM changes presently.

I would have to look at that code again, but when I did the first pass a few months ago, it did not seem prohibitively complex to perform that logic.

It would also be possible to add the VM changes at a later date so long as the state for commission rate is present. Status quo would be to ignore this bit of state in the VM (which is no worse than current state).